x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/all_rules/actions.tsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  deleteRules,
+  duplicateRules,
+  enableRules,
+} from '../../../../containers/detection_engine/rules/api';
+import { Action } from './reducer';
+import { Rule } from '../../../../containers/detection_engine/rules/types';
+
+export const editRuleAction = () => {};
+
+export const runRuleAction = () => {};
+
+export const duplicateRuleAction = async (
+  rule: Rule,
+  dispatch: React.Dispatch<Action>,
+  kbnVersion: string
+) => {
+  dispatch({ type: 'updateLoading', ids: [rule.id], isLoading: true });
+  const duplicatedRule = await duplicateRules({ rules: [rule], kbnVersion });
+  dispatch({ type: 'updateLoading', ids: [rule.id], isLoading: false });
+  dispatch({ type: 'updateRules', rules: duplicatedRule });
+};
+
+export const exportRulesAction = async (rules: Rule[], dispatch: React.Dispatch<Action>) => {
+  dispatch({ type: 'setExportPayload', exportPayload: rules });
+};
+
+export const deleteRulesAction = async (
+  ids: string[],
+  dispatch: React.Dispatch<Action>,
+  kbnVersion: string
+) => {
+  dispatch({ type: 'updateLoading', ids, isLoading: true });
+  const deletedRules = await deleteRules({ ids, kbnVersion });
+  dispatch({ type: 'deleteRules', rules: deletedRules });
+};
+
+export const enableRulesAction = async (
+  ids: string[],
+  enabled: boolean,
+  dispatch: React.Dispatch<Action>,
+  kbnVersion: string
+) => {
+  try {
+    dispatch({ type: 'updateLoading', ids, isLoading: true });
+    const updatedRules = await enableRules({ ids, enabled, kbnVersion });
+    dispatch({ type: 'updateRules', rules: updatedRules });
+  } catch {
+    dispatch({ type: 'updateLoading', ids, isLoading: false });
```

Comment:
Optional:

That's an interesting `catch`. I haven't seen one without an exception object before. Didn't know you could do that.

I would add a comment about when/why you are getting an exception? I am guessing it is because of the `await eanbledRules` at the moment.

Reply:
The ole `quick catch`. 😉 This should throw up a toast with the error once we add support for that. I'll need to loop back and make sure all the actions have a supported error flow with our good friend `dispatchToaster`. Will add a `TODO` comment as suggested for tracking.